crates/ruff_linter/src/rules/pyupgrade/rules/pep695.rs

```diff
@@ -0,0 +1,241 @@
+//! Shared code for [`use_pep695_type_alias`] (UP040) and [`use_pep695_type_parameter`] (UP046)
+
+use std::fmt::Display;
+
+use ruff_python_ast::{
+    self as ast,
+    visitor::{self, Visitor},
+    Expr, ExprCall, ExprName, ExprSubscript, Identifier, Stmt, StmtAssign, TypeParam,
+    TypeParamParamSpec, TypeParamTypeVar, TypeParamTypeVarTuple,
+};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::{Ranged, TextRange};
+
+pub(crate) use use_pep695_type_alias::*;
+pub(crate) use use_pep695_type_parameter::*;
+
+mod use_pep695_type_alias;
+mod use_pep695_type_parameter;
+
+#[derive(Debug)]
+enum TypeVarRestriction<'a> {
+    /// A type variable with a bound, e.g., `TypeVar("T", bound=int)`.
+    Bound(&'a Expr),
+    /// A type variable with constraints, e.g., `TypeVar("T", int, str)`.
+    Constraint(Vec<&'a Expr>),
+}
+
+#[derive(Copy, Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
```

Comment:
nit: with the latest version of the PR, I think this enum no longer needs the `Ord` and `PartialOrd` implementations (and to me there's no "obvious" order they should always go in, so I'd prefer to remove them)

```suggestion
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Reply:
Good call, I mean to make a comment about the ordering, but you've saved me from that too!